### PR TITLE
Add Terraform configuration for Notary infrastructure on Azure

### PR DESCRIPTION
**Pull request by Duffney**
A Terraform module named `notary` has been added to the iac folder with an accompanying doc that explains the process for using the configuration to deploy and configure the necessary Azure resources to use notation to sign container images in ACR.

Also, I minor change as made to the `setup.sh` file to resolve a change in parameter names within the AzureCLI.

### CODE REVIEW

File: iac/notary/modules/configure/installnotation.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export PATH="$HOME/bin:$PATH"
+
+# Install Azure Keyvault Plugin
+
+# Create a directory for the plugin
+[ -d ~/.config/notation/plugins/azure-kv ] || mkdir -p ~/.config/notation/plugins/azure-kv
+
+# Download the plugin
+curl -Lo notation-azure-kv.tar.gz \
+    https://github.com/Azure/notation-azure-kv/releases/download/v0.1.0-alpha.1/notation-azure-kv_0.1.0-alpha.1_${osVersion}_amd64.tar.gz > /dev/null 2>&1
```
Comment (dtzar) [inline iac/notary/modules/configure/installnotation.sh:36]: This is a very old version of the plugin.

File: iac/notary/modules/configure/installnotation.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# Choose a binary
+timestamp=20220121081115
+commit=17c7607
+
+# Check if ~/bin is there, if not create it 
+[ -d ~/bin ] || mkdir ~/bin
+
+# Download Notation from pre-release
+curl -Lo notation.tar.gz https://github.com/notaryproject/notation/releases/download/feat-kv-extensibility/notation-feat-kv-extensibility-$timestamp-$commit.tar.gz > /dev/null 2>&1
```
Comment (dtzar) [inline iac/notary/modules/configure/installnotation.sh:14]: This also is a very old version of notation you should not be using.

File: iac/notary/modules/deploy/main.tf [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        days_before_expiry = 30
+      }
+    }
+
+    secret_properties {
+      content_type = "application/x-pkcs12"
+    }
+
+    x509_certificate_properties {
+      extended_key_usage = ["1.3.6.1.5.5.7.3.1","1.3.6.1.5.5.7.3.2","1.3.6.1.5.5.7.3.3"]
```
Comment (dtzar) [inline iac/notary/modules/deploy/main.tf:222]: Please only use the codesign EKU.  This is updated in the public azure doc.

**Review by dtzar — CHANGES_REQUESTED** [review]

File: iac/notary/modules/deploy/main.tf (hunk 3 above)
Comment (Duffney) [reply iac/notary/modules/deploy/main.tf:222]: Just to confirm the codesign EKU is `1.3.6.1.5.5.7.3.3` correct? I couldn't find any docs specific to Azure, but I did find this https://ldapwiki.com/wiki/ExtendedKeyUsage.

**Comment by Duffney** [comment]
@dtzar I'm working to resolve all these issues, I should have an update shortly. :)

File: iac/notary/modules/deploy/main.tf (hunk 3 above)
Comment (dtzar) [reply iac/notary/modules/deploy/main.tf:222]: Correct.  Referenced here: https://learn.microsoft.com/en-us/azure/container-registry/container-registry-tutorial-sign-build-push#create-a-self-signed-certificate-azure-cli

Comment (Duffney) [reply iac/notary/modules/deploy/main.tf:222]: Thanks! I've updated the code to only use the code sign EKU.

**Pull request merged**